### DISCUSSION TITLE
[Syntax Error?] What is "!"?

### DISCUSSION THREAD

**Discussion by vadya-coder** [reduxjs/react-redux#2207] [Q&A]
Apparently, I do not know js. What is "!"?
https://github.com/reduxjs/react-redux/blob/56dbf28ea74fe50c7c6a06b7ea0c78018fa48f4b/src/hooks/useReduxContext.ts#L22

**Comment by EskiMojo14** [comment]
it's typescript specific, it's a non-nullish assertion. Basically it's the equivalent to `return contextValue as NonNullable<typeof contextValue>`
